runner/src/main/scala/com/github/dwickern/swagger/SwaggerRunner.scala

```diff
@@ -4,18 +4,26 @@ import io.swagger.converter.ModelConverters
 import io.swagger.parser.SwaggerParser
 import io.swagger.util.Json
 import org.slf4j.LoggerFactory
-import play.api.{Environment, Mode}
+import play.api.{Configuration, Environment, Mode}
 import play.modules.swagger.SwaggerPluginImpl
 
 import java.io.File
+import java.util.{Map => JMap}
+
+import scala.jdk.CollectionConverters._
 
 object SwaggerRunner {
   private lazy val logger = LoggerFactory.getLogger(getClass)
 
-  def run(rootPath: File, host: String, validate: Boolean): String = {
+  // configuration should be an Option, but it does not play nice with reflective loading, so we use Java-style nullable map instead
+  def run(rootPath: File, host: String, validate: Boolean, configuration: JMap[String, Any]): String = {
     val classLoader = getClass.getClassLoader
     val env = Environment(rootPath, classLoader, Mode.Dev)
-    val conf = play.api.Configuration.load(env)
+    val conf = if (configuration == null) {
+      Configuration.load(env)
+    } else {
+      Configuration.from(defaultConfiguration ++ configuration.asScala)
```

Comment:
Instead could we use the [reference configuration](https://github.com/dwickern/swagger-play/blob/master/src/main/resources/reference.conf) as a fallback? That would seem more resilient to upstream changes in swagger-play, and the behavior would more closely match `application.conf`-based configuration.

```suggestion
      Configuration.reference ++ Configuration.from(configuration.asScala.toMap)
```

Reply:
I did not know about that reference configuration, it is an improvement over the manual map indeed.